### ISSUE TITLE
Allow for 'general' options to be set in the gatsby-plugin-google-analytics config.

### ISSUE THREAD

**Issue by tinder-kyleboss** [gatsbyjs/gatsby#15168]
## Summary

It would be nice to allow users of the gatsby-plugin-google-analytics plugin to be able to set ['general' field values](https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#general) in the configuration section, very similar to what we have with the ['create only' fields](https://github.com/gatsbyjs/gatsby/pull/7316). 😸 

### Basic example

```
{
      resolve: 'gatsby-plugin-google-analytics',
      options: {
          trackingId: 'XXX',
          head: false,
          anonymize: true,
          respectDNT: false,
          transport: 'beacon'
      }
}
```

### Motivation

I wanted to be abled to use some of the general google analytics features for my own project and was about to create my own solution. I would like to know if this is something the community in general would be interested in or hear about reasons this wasn't included in the first place (if any, of course 😅)

Thanks fam! 🙏

**Comment by prestonso** [comment]
Hi @tinder-kyleboss!

Thanks for opening this issue and for your feature request. I've labeled this issue with `help wanted` to signal to interested community members that they can contribute the added functionality to set those `general` field values.

Would you like to open a pull request contributing this feature? We'd be more than happy to review it!

Thanks for using Gatsby! 💪 💜

**Comment by tinder-kyleboss** [comment]
Cool! Whipping up a PR now!

**Comment by tinder-kyleboss** [comment]
Not stale :) Working on the fix (here)[https://github.com/gatsbyjs/gatsby/pull/15280]

**Comment by tinder-kyleboss** [comment]
Fixed by #15280. Thanks everyone!
